boutdata/restart.py

```diff
@@ -632,7 +632,13 @@ def redistribute(npes, path="data", nxpe=None, output=".", informat=None, outfor
                 if yindex_global + myg >= iy*mysub and yindex_global + myg < (iy+1)*mysub+2*myg:
                     outfile.write(v, data[ix*mxsub:(ix+1)*mxsub+2*mxg, :])
                 else:
-                    nullarray = BoutArray(np.zeros([mxsub+2*mxg, mysub+2*myg]), attributes={"bout_type":"FieldPerp", "yindex_global":-myg-1})
+                    nullarray = BoutArray(
+                        np.zeros([mxsub + 2 * mxg, mz]),
```

Comment:
Sorry, I actually already fixed this bug in #19, but that hasn't been merged yet. @ZedThree if you have time to give #19 a look-over, I could fix the merge conflicts in that. If not, it might be better to merge this quickly to fix the bug.

For consistency/future-proofing, I think `mz` should be `mzsub`
```suggestion
                        np.zeros([mxsub + 2 * mxg, mzsub]),
```

Reply:
I thought it looked familiar! Ok, let's get #19 in